### PR TITLE
[TIR] Fix predefined inverse map in layout transform dtype legalization

### PR DESCRIPTION
**Pull request by vinx13**
`LegalizeIndexMapDType` previously ignore pre-defined inverse map, which causes error in `TransformLayout` when the inverse index map is predefined but the arithmetic analysis fails to compute the inverse map.

cc @masahi @Lunderberg

### CODE REVIEW

**Review by junrushao — APPROVED** [review]
LGTM!

File: src/tir/schedule/primitive/layout_transformation.cc [hunk 1]
```diff
@@ -1079,7 +1079,13 @@ IndexMap LegalizeIndexMapDType(const IndexMap& index_map, const Buffer& buf) {
       return SubstituteWithDataTypeLegalization(index,
                                                 [&](const Var& var) { return var_map.Get(var); });
     });
-    return IndexMap(initial_indices, final_indices);
+    Optional<IndexMap> opt_inverse_index_map =
+        Downcast<Optional<IndexMap>>(index_map->inverse_index_map);
+    if (opt_inverse_index_map.defined()) {
+      IndexMap inverse_index_map = Downcast<IndexMap>(opt_inverse_index_map.value());
+      opt_inverse_index_map = LegalizeIndexMapDType(inverse_index_map, final_indices);
```
Comment (masahi) [inline src/tir/schedule/primitive/layout_transformation.cc:1086]: Probably we can remove `Downcast<IndexMap>` at L1083 and L1085?

Comment (vinx13) [reply src/tir/schedule/primitive/layout_transformation.cc]: updated, I'm keeping L1083 as `inverse_index_map` is defined as `Optional<ObjectRef>`

File: src/tir/schedule/primitive/layout_transformation.cc [hunk 2]
```diff
@@ -1079,7 +1079,12 @@ IndexMap LegalizeIndexMapDType(const IndexMap& index_map, const Buffer& buf) {
       return SubstituteWithDataTypeLegalization(index,
                                                 [&](const Var& var) { return var_map.Get(var); });
     });
-    return IndexMap(initial_indices, final_indices);
+    Optional<IndexMap> opt_inverse_index_map =
+        Downcast<Optional<IndexMap>>(index_map->inverse_index_map);
+    if (opt_inverse_index_map.defined()) {
+      opt_inverse_index_map = LegalizeIndexMapDType(opt_inverse_index_map.value(), final_indices);
```
Comment (masahi) [inline src/tir/schedule/primitive/layout_transformation.cc:1085]: Can we guarantee that this recursive call would always bottom out?

**Pull request merged**